### ISSUE TITLE
Upgrade from C++03 to C++11 (low priority)

### ISSUE THREAD

**Issue by fritzo** [nimble-dev/nimble#490]
## Why?

C++11 features will allow us to simplify code on the C++ side.

## How?

This may be as easy as adding `-std=c++0x` to `PKG_CPPFLAGS` in `configure` and `configure.win`.
See [bigmemory](https://github.com/kaneplusplus/bigmemory/commit/d6690b28)'s migration commit for example.

**Comment by perrydv** [comment]
@paciorek Can we try turning on C++11 in 0.6-7?  It would help to be able to use it soon, but it might be a good idea to turn it for a release before we're heavily dependent on it.

Here is the [relevant section of Writing R Extensions](https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Using-C_002b_002b11-code).

It looks like a small modification to configure and various Makevars files.  I can make a PR if this sounds ok.

**Comment by paciorek** [comment]
sure.

**Comment by paciorek** [comment]
I was going to do a trial rhub check well in advance of release, but now I'm going to wait on this PR since it will affect things that have in the past caused CRAN issues.

**Comment by perrydv** [comment]
I just files a PR to turn on C++11.
